woocommerce/woocommerce-android
# Issue/2910 batch add product tags

Fixes #2190  by adding logic to add new tags to a product.

~**Note that this PR is in draft this till [bottom sheet PR](https://github.com/woocommerce/woocommerce-android/pull/2617) can be reviewed and merged.**~

### Changes
- When a tag is entered, the logic is to first check if this tag exists in the local db. (We are fetching 100 tags at a time so most of the tags should be available in the local db). 
  - If the tag exists in the local db, then the tag is added automatically to the product. 
  - If the tag does not exist in the local db, we can assume that the tag added is new, so we add it to a new list. When the user clicks on the "DONE" button, if this list is not empty, we first call an API to add these tags to the site. If the response is successful, the tag is automatically added to the product. If the response results in an error, a generic snackbar is displayed. 

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/87903778-6eaeed00-ca7a-11ea-9458-181c921ec1c7.gif" width="300"/>

#### Testing
- Click on a product from the Products list (without any tags).
- Click on the bottom sheet and click on `Tags`.
- Type a new tag in the text field. Notice the tag is displayed correctly and the "DONE" menu button is displayed.
- Click on the "DONE" menu button and notice that the add is added successfully to the site and that you are redirected to the Product detail screen.
- Notice the new tags are displayed in the Product detail screen.
- Click on the UPDATE button and verify that the tags are updated successfully for the product.
- Click on the Tags section for the same product.
- Remove the selected tags and click on DONE button.
- Notice the tags section is no longer visible. 
- Click on the UPDATE menu button.
- Verify that the tags are updated successfully for the product.

Update release notes:
